src/main/java/seedu/address/logic/Pomodoro.java

```diff
@@ -0,0 +1,163 @@
+        this.timerLabel = timerLabel;
+    }
+
+    public void start(float timeInMinutes) {
+        startTime = (int)(timeInMinutes * 60);
+        timeSeconds = new SimpleIntegerProperty(startTime);
+        configureUi();
+        configureTimer();
+    }
+
+    public void pause() throws NullPointerException {
+        try {
+            timeline.pause();
+        } catch (NullPointerException ne) {
+            throw ne;
+        }
+    }
+
+    public void unpause() throws NullPointerException {
+        try {
+            timeline.play();
+        } catch (NullPointerException ne) {
+            throw ne;
+        }
+    }
+
+    private void configureUi() {
+        timerLabel.textProperty().bind(Bindings.createStringBinding(() -> {
+            if (timeSeconds.getValue() == null) {
+                return "";
+            } else {
+                int secondsRemaining = timeSeconds.get();
+                int minutePortion = secondsRemaining / 60;
+                int secondPortion = secondsRemaining % 60;
+                return String.format("%02d:%02d", minutePortion, secondPortion);
+            }
+        }, timeSeconds));
+    }
+
+    private void configureTimer() {
+        if (timeline != null) {
+            timeline.stop();
+        }
+        timeSeconds.set(startTime);
+        timeline = new Timeline();
+        timeline.getKeyFrames().add(
+            new KeyFrame(Duration.seconds(startTime + 1),
+            new KeyValue(timeSeconds, 0)));
+        timeline.playFromStart();
+        timeline.setOnFinished(event -> {
+            this.setPromptState(PROMPT_STATE.CHECK_DONE);
+            resultDisplay.setFeedbackToUser(CHECK_DONE_MESSAGE);
+        });
+    }
+
+    public PROMPT_STATE getPromptState() {
+        return this.promptState;
+    }
+
+    public void setPromptState(PROMPT_STATE promptState) {
+        this.promptState = promptState;
+    }
+
+    public void checkBreakActions() {
+        this.setPromptState(PROMPT_STATE.CHECK_TAKE_BREAK);
+    }
+
+    public void takeABreak() {
+        if (timeline != null) {
+            timeline.stop();
+        }
+        timeSeconds.set(restTime);
+        timeline = new Timeline();
+        timeline.getKeyFrames().add(
+            new KeyFrame(Duration.seconds(restTime + 1),
+            new KeyValue(timeSeconds, 0)));
+        timeline.playFromStart();
+        timeline.setOnFinished(event -> {
+            resultDisplay.setFeedbackToUser("Breaks over! What shall we do next?");
+            this.setPromptState(PROMPT_STATE.NONE); // App back to neutral
+        });
+    }
+
+    public void setDoneParams(Model model, List<Task> originList, int taskIndex) {
+        this.model = model;
+        this.originList = originList;
+        this.taskIndex = taskIndex;
+    }
+
+    private void clearDoneParams() {
+        this.model = null;
+        this.originList = null;
+        this.taskIndex = -1;
+    }
+
+    public void doneTask() {
```

Comment:
Not sure whether these different definitions of done task will be confusing down the road, maybe add some comments to clarify the difference